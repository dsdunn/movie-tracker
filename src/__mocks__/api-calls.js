export const signUp = jest.fn().mockImplementation((state) => {
  if(state.name === 'bugs') {
    return Promise.resolve({id: 1})
  } else {
    return 'nothing'
  }
}); 
