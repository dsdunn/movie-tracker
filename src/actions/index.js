export const addMovies = (movies) => ({
  type: "ADD_MOVIES",
  movies
});

export const toggleFav = (id) => {
  type: "TOGGLE_FAV",
  id
};

export const toggleFilter = () => {
  type: "TOGGLE_FILTER"
};