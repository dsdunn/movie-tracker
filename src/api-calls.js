import { apiKey } from './apiKey';

export const getMovies = async () => {
  try {
    const url = `https://api.themoviedb.org/3/discover/movie?api_key=${apiKey}&language=en-US&sort_by=popularity.desc&include_adult=false&include_video=false&page=1&primary_release_year=2018`;
    const response = await fetch(url)
    return response.json();
  } 
  catch(error) {
    return error.message;
  }
}

export const logIn = async (state) => {
  try{
    const url = 'http://localhost:3000/api/users/';
    const {email, password} = state;

    const response = await fetch(url, {
      method: 'POST',
      body: JSON.stringify({email, password}),
      headers:{
        'Content-Type': 'application/json'
      }
    })
    const result = await response.json();
    return result.data;
 
  } 
  catch (error) {
    return false;
  }
}

export const signUp = async (state) => {
  try{
    const url = 'http://localhost:3000/api/users/new';
    const {name, email, password} = state;

    const response = await fetch(url, {
      method: 'POST',
      body: JSON.stringify({name, email, password}),
      headers:{
        'Content-Type': 'application/json'
      }
    });
    return await response.json();
  } 
  catch (error) {
    return false;
  }
}

export const postFavorite = async (favorite) => {
  console.log(favorite)
  try {
    const url = 'http://localhost:3000/api/users/favorites/new';
    const response = await fetch(url, {
      method: 'POST',
      body: JSON.stringify(favorite),
      headers: {
        'Content-Type': 'application/json'
      } 
    });
    return await response.json();
  }
  catch (error) {
    return false;
  }
}
















