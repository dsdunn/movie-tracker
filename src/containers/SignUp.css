.sign-up {
  position: absolute;
  display: flex;
  top: 30%;
  left: 35%;
  padding: 1em;
  border-radius: 1em;
  flex-direction: column;
  background-color: rgba(33, 33, 33, .8);
}

.sign-up input {
  padding: .3em 1em;
  font-size: .7em;
  margin: .2em 0;
  border-radius: 8px;
  border: 0;
  font-size: 1.5em;
}

.sign-up button {
  width: 20em;
  border: 0;
  border-radius: 8px;
  margin: .2em 0;
  font-size: 1.5em;
}

.user-message {
  font-size: 1.5em;
  margin: 0 .3em;
  color: #CFD8DC;
}