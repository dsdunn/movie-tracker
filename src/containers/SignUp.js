import React, { Component } from 'react';
import { connect } from 'react-redux';
import { signUp } from '../api-calls';
import { loginUser } from '../actions';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import './SignUp.css';

export class SignUp extends Component {
  constructor(props) {
    super(props);
    this.state={
      name: '',
      email: '',
      password: '',
      hasError: false
    };
  }

  handleChange = (input) => {
    const { name, value } = input.target;
    this.setState({
      [name]: value
    });
  }

  handleSubmit = async (event) => {
    event.preventDefault();
    const reply = await signUp(this.state);
    if (reply.id) {
      this.setState({id: reply.id, favorites: [], hasError: false});
      this.props.login(this.state);    
      this.props.history.push('/');
    } else {
      this.setState({ hasError: true });
    }
  }

  handleError = () => {
    if (this.state.hasError) {
      return (
        <h5 className="user-message">Try A Diffrent E-Mail</h5>
      );
    }
  }

  render() {
    return (
      <form className='sign-up' onSubmit={this.handleSubmit}>
        <input 
          id='name' 
          name='name' 
          placeholder='Name'
          onChange={this.handleChange}
        />
        <input 
          id= 'email' 
          type='email' 
          name='email'
          placeholder='E-Mail'
          onChange={this.handleChange}
        />
        <input 
          id='password' 
          type='password' 
          name='password'
          placeholder='Password'
          onChange={this.handleChange}
        />
        <button onClick={this.showSignUp}>Sign Up</button>
        <Link to='/' ><button>Home</button></Link>
        {this.handleError()}
      </form>
    );
  }
}

export const mapStateToProps = (state) => ({
  user: state.user
});

export const mapDispatchToProps = (dispatch) => ({
  login: (user) => dispatch(loginUser(user))
});

SignUp.propTypes = {
  login: PropTypes.func,
  user: PropTypes.object,
  history: PropTypes.object
};

export default connect(mapStateToProps, mapDispatchToProps)(SignUp);