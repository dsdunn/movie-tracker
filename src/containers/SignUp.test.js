import React from 'react';
import { shallow } from 'enzyme';
import  { SignUp, mapStateToProps, mapDispatchToProps } from './SignUp';
import { signUp } from '../api-calls';
import { loginUser, signUpUser } from '../actions';

jest.mock('../api-calls.js');

describe.only('SignUp', () => {
  let wrapper, mockEvent, mockState;
  beforeEach(() => {
    wrapper = shallow(<SignUp/>);
    mockEvent = {preventDefault: jest.fn(),target:{name: "name", value: "bob"}};
    mockState = {
      name: '',
      email: '',
      password: '',
      hasError: false
    }
  })

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should instantiate with some state', () => {
    expect(wrapper.instance().state).toEqual(mockState);
  })

  it('should update state when input changes', () => {
    wrapper.find('#name').simulate('change', mockEvent);
    
    expect(wrapper.state('name')).toEqual('bob');
  })

  it('should call signUp on submit', () => {
    wrapper = shallow(<SignUp login={ jest.fn()} history={['yeah']}/>) 

    wrapper.find('.sign-up').simulate('submit', mockEvent);
    expect(signUp).toBeCalledWith(mockState);
  })

  it('should set hasError to true on failed response',async () => {
    wrapper = shallow(<SignUp login={ jest.fn()} history={['yeah']}/>)
    
    await wrapper.find('.sign-up').simulate('submit', mockEvent);
    expect(wrapper.state('hasError')).toEqual(true)
  })

  it('should set hasError to false on successful response',async () => {
    wrapper = shallow(<SignUp login={ jest.fn()} history={['yeah']}/>)
    wrapper.setState({name: 'bugs'})

    await wrapper.find('.sign-up').simulate('submit', mockEvent);
    expect(wrapper.state('hasError')).toEqual(false)
  })

  describe('mapStateToProps', () => {
    it('should return an object with a user', () => {
      const mockState = {
        user: {name: 'Joe'},
        dog: {breed: 'mut'}
      }
      const expectedProps = {user: {name: 'Joe'}};
      const mappedProps = mapStateToProps(mockState);

      expect(mappedProps).toEqual(expectedProps)
    })
  })

  describe('mapDispatchToProps', () => {
    it('calls dispatch with loginUser when handleSubmit is called', () => {
      const mockDispatch = jest.fn();
      const mockUser = {name: 'Joe', password: 'bob'}
      const actionToDispatch = loginUser(mockUser);
      const mappedProps = mapDispatchToProps(mockDispatch);

      mappedProps.login(mockUser);

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })
  })

})






