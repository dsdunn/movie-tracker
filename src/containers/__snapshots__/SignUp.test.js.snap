// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignUp />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label
          htmlFor="name"
>
          Name:
</label>,
        <input
          id="name"
          name="name"
          onChange={[Function]}
/>,
        <label
          htmlFor="email"
>
          E-Mail:
</label>,
        <input
          id="email"
          name="email"
          onChange={[Function]}
          type="email"
/>,
        <label
          htmlFor="password"
>
          Password
</label>,
        <input
          id="password"
          name="password"
          onChange={[Function]}
          type="password"
/>,
        <button
          onClick={undefined}
>
          Sign Up
</button>,
        <Link
          replace={false}
          to="/"
>
          <button>
                     Home 
          </button>
</Link>,
        undefined,
      ],
      "className": "sign-up",
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Name:",
          "htmlFor": "name",
        },
        "ref": null,
        "rendered": "Name:",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "id": "name",
          "name": "name",
          "onChange": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "E-Mail:",
          "htmlFor": "email",
        },
        "ref": null,
        "rendered": "E-Mail:",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "id": "email",
          "name": "email",
          "onChange": [Function],
          "type": "email",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Password",
          "htmlFor": "password",
        },
        "ref": null,
        "rendered": "Password",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "id": "password",
          "name": "password",
          "onChange": [Function],
          "type": "password",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Sign Up",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": "Sign Up",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <button>
             Home 
</button>,
          "replace": false,
          "to": "/",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " Home ",
          },
          "ref": null,
          "rendered": " Home ",
          "type": "button",
        },
        "type": [Function],
      },
      undefined,
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label
            htmlFor="name"
>
            Name:
</label>,
          <input
            id="name"
            name="name"
            onChange={[Function]}
/>,
          <label
            htmlFor="email"
>
            E-Mail:
</label>,
          <input
            id="email"
            name="email"
            onChange={[Function]}
            type="email"
/>,
          <label
            htmlFor="password"
>
            Password
</label>,
          <input
            id="password"
            name="password"
            onChange={[Function]}
            type="password"
/>,
          <button
            onClick={undefined}
>
            Sign Up
</button>,
          <Link
            replace={false}
            to="/"
>
            <button>
                         Home 
            </button>
</Link>,
          undefined,
        ],
        "className": "sign-up",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Name:",
            "htmlFor": "name",
          },
          "ref": null,
          "rendered": "Name:",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "id": "name",
            "name": "name",
            "onChange": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "E-Mail:",
            "htmlFor": "email",
          },
          "ref": null,
          "rendered": "E-Mail:",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "id": "email",
            "name": "email",
            "onChange": [Function],
            "type": "email",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Password",
            "htmlFor": "password",
          },
          "ref": null,
          "rendered": "Password",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "id": "password",
            "name": "password",
            "onChange": [Function],
            "type": "password",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sign Up",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": "Sign Up",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <button>
               Home 
</button>,
            "replace": false,
            "to": "/",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " Home ",
            },
            "ref": null,
            "rendered": " Home ",
            "type": "button",
          },
          "type": [Function],
        },
        undefined,
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
